File: easy/Arrays/validate-subsequence/solution-1.js
array = [5, 1, 22, 25, 6, -1, 8, 10]
sequence = [1, 6, -1, 10]

function isValidSubsequence(array, sequence) {
const subSequence = []

  for(item of array) {
    if (sequence.includes(item)) {
      subSequence.push(item)
    }
  }
  return sequence.every((item, index) => item == subSequence[index]);
}

isValidSubsequence(array, sequence)

File: easy/Arrays/validate-subsequence/solution-1.py
def isValidSubsequence(array, sequence):
    arrayIndex = 0
    sequenceIndex = 0

    while arrayIndex < len(array) and sequenceIndex < len(sequence):
        if array[arrayIndex] == sequence[sequenceIndex]:
            sequenceIndex += 1
        arrayIndex += 1

        
    return sequenceIndex == len(sequence)
